.github/workflows/deny_dirty_cargo_locks.yml

```diff
@@ -0,0 +1,32 @@
+name: Check no Cargo.lock files are dirty
+
+on:
+  [push, pull_request]
+
+jobs:
+  no_dirty_cargo_locks_check:
+    runs-on: ubuntu-latest
+    steps:
+      - name: "Checkout repository"
+        uses: actions/checkout@v3
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
+
+      - name: "Check no Cargo.lock files are dirty" 
```

Comment:
This seems more complex than needed, 1 lock file should be generated for the whole workspace, so just doing `cargo generate-lockfile --locked` should be sufficient, am I missing something?

Reply:
There are multiple lockfiles (for example the ones updated in this PR for the uffd example and the seccomp demo). Also, `generate-lockfile` implicitly performs a `cargo update` since it generates a lockfile from scratch, which is why this Action is using `cargo metadata`.

Reply:
Under https://doc.rust-lang.org/cargo/commands/cargo-generate-lockfile.html#manifest-options it notes:

> --frozen
> --locked
>
>  Either of these flags requires that the Cargo.lock file is up-to-date. If the lock file is missing, or it needs to be updated, Cargo will exit with an error. The --frozen flag also prevents Cargo from attempting to access the network to determine if it is out-of-date.
> 
>  These may be used in environments where you want to assert that the Cargo.lock file is up-to-date (such as a CI build) or want to avoid network access.

Specifically 

> If the lock file is missing, or it needs to be updated, Cargo will exit with an error.

Do we know what constitutes `needs to be updated`?

Is this when:

1. `Cargo.lock` is incompatible with the `Cargo.toml`s such that it must be updated to be made valid.
2. Dependencies in `Cargo.lock` can be updated (e.g. `cargo update` would produce a change).

My understand was that this related to the 1st point.

Reply:
In this case, its the latter (generate-lockfile --locked fails against our main Cargo.lock for example, even though it is in-sync with Cargo.toml)